Chat/WebApi/Controllers/MessagesController.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Application.Dtos;
+using Application.Extentions;
+using Application.Ports.Services;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace WebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    [Authorize]
+    public class MessagesController : ControllerBase
+    {
+        private readonly IMessageService _messageService;
+        
+
+        public MessagesController(IMessageService messageService)
+        {
+            _messageService = messageService;
+            
+        }
+
+        
+
+        [HttpPut("{chatid}/{id}/status")]
+        public async Task<ActionResult> ChangeMessageStatusAsync([FromRoute] int chatid, [FromRoute] int id, [FromBody] string status,CancellationTokenSource _source)
```

Comment:
Your method should accept CancellationToken, and remove the underscore (_) in the name